src/utils/observability/src/config.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug)]
+pub struct Config {
+    /// Name of the service that will appear e.g. in OTEL traces
+    pub service_name: String,
+    /// Version of the service that will appear e.g. in OTEL traces
+    pub service_version: String,
+    /// Log levels that will be used if `RUST_LOG` was not specified explicitly
+    pub default_log_levels: String,
+    /// OpenTelemetry protocol endpoint to export traces to
+    pub otlp_endpoint: Option<String>,
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            service_name: "unnamed-service".to_string(),
+            service_version: "0.0.0".to_string(),
+            default_log_levels: "info".to_string(),
+            otlp_endpoint: None,
+        }
+    }
+}
+
+impl Config {
+    pub fn from_env() -> Self {
+        let mut cfg = Self::default();
+        if let Some(service_name) = std::env::var("SERVICE_NAME").ok().filter(|v| !v.is_empty()) {
```

Comment:
**Q:** What about envvar prefix?

Reply:
good point, I will add optional prefix to give ability to disambiguate and avoid potential collision with k8s env vars

Reply:
done!